pydicom/filewriter.py

```diff
@@ -144,15 +150,22 @@ def correct_ambiguous_vr(ds, is_little_endian):
     ds : pydicom.dataset.Dataset
         The corrected dataset
     """
+    # if we want to write with the same endianess and VR handling as
+    # the read dataset we want to preserve raw data elements
+    if ds.write_like_original():
+        dataset_elements = ds.elements()
+    else:
+        dataset_elements = ds
+
     # Iterate through the elements
-    for elem in ds:
-        # Iterate the correction through any sequences
-        if elem.VR == 'SQ':
+    for elem in dataset_elements:
+        # raw data element sequences can be written as they are, because we
+        # have ensured that the transfer syntax has not changed at this point
+        if elem.VR == 'SQ' and not elem.is_raw:
```

Comment:
If `ds.write_like_original()` is `True` then would you even need to do the VR correction? It might be better to change `filewriter.write_dataset`.

Reply:
I will check this.

Reply:
Ok, you are write - changed this.